src/main/cc/common_cpp/protobuf_util/riegeli_io.h

```diff
@@ -67,18 +66,23 @@ absl::Status ReadRiegeliFile(absl::string_view path,
 // Return error status if the writing fails.
 // The details about the format of the output file can be found in
 // https://github.com/google/riegeli/blob/master/doc/riegeli_records_file_format.md
+// Any message in @messages is packed to google::protobuf::Any before written to
+// the file.
 template <typename Message>
 absl::Status WriteRiegeliFile(absl::string_view path,
                               const std::vector<Message>& messages) {
-  riegeli::RecordWriter writer(riegeli::FdWriter(path, O_WRONLY | O_CREAT));
+  riegeli::RecordWriter writer(
+      riegeli::FdWriter(path, O_WRONLY | O_CREAT | O_TRUNC));
+  google::protobuf::Any any;
   for (const Message& message : messages) {
-    if (!writer.WriteRecord(message)) {
-      if (!writer.Close()) {
-        return writer.status();
-      }
-      return absl::InvalidArgumentError(
-          absl::StrCat("Unable to write message to file: ", path));
+    if (!any.PackFrom(message)) {
+      return absl::InternalError("Fails to serialize the input message.");
+    }
+    if (!writer.WriteRecord(any)) {
+      writer.Close();
```

Comment:
There is no need to close RecordWriter on error. File contents are likely truncated anyway.